programs/fileio.c

```diff
@@ -1300,7 +1300,10 @@ FIO_compressZstdFrame(FIO_ctx_t* const fCtx,
         int windowLog;
         UTIL_HumanReadableSize_t windowSize;
         CHECK(ZSTD_CCtx_getParameter(ress.cctx, ZSTD_c_windowLog, &windowLog));
-        if (windowLog == 0) {
+        if (prefs->ldmFlag) {
+            /* If long mode is set libzstd will set this window size internally */
+            windowLog = ZSTD_WINDOWLOG_LIMIT_DEFAULT;
```

Comment:
`--long` mode can be activated with a parameter.
For example, `--long=30` means "use a 1 GB window size".
The shorter `--long` is equivalent to `--long=27`.

This means we can't guarantee that the window size is necessarily `ZSTD_WINDOWLOG_LIMIT_DEFAULT` when the `--long` mode is activated, it could be a different value if an explicit parameter was passed.

Reply:
Thanks, yes, there's an override via `ZSTD_overrideCParams` that I missed, so the ldmFlag check should only be performed if the windowLog ends up as 0.